src/redux/actions/viewSingleArticle/getRecentArticles.js

```diff
@@ -0,0 +1,20 @@
+import * as actions from '../actionTypes';
+
+export const displayArticleActionCreator = (data) => ({
+  type: actions.FETCH_ARTICLES_ON_LANDING_PAGE_SUCCESS,
+  payload: data,
+});
+export const displayArticleActionError = (error) => ({
+  type: actions.FETCH_ARTICLES_ON_LANDING_PAGE_ERROR,
+  payload: error,
+});
+
+export const getAllArticles = () => (dispatch) => fetch(`${actions.BASE_URL}/api/${actions.VERSION}/articles`)
+  .then((res) => res.json())
+  .then((data) => {
+    if (data.status === 200) {
+      return dispatch(displayArticleActionCreator(data));
+    }
+    return dispatch(displayArticleActionError(data));
+  })
+  .catch((error) => `server error : ${error}`);
```

Comment:
how about retrieving article then sort it in your component as it is going to come from most recent

Reply:
Nice suggestion Salvi. I think we should prioritize the motivation as to why in the first place we have two separate API endpoints. i.e endpoint for getting all articles and getting a single articles. I used the logic above to separate concerns.